src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfCodec.java

```diff
@@ -0,0 +1,376 @@
+            // GencodeGtfFeature.
+            if ((gene != null) && (transcript != null) && (featureType.equals("gene") || featureType.equals("transcript") )) {
+
+                aggregateRecordsIntoGeneFeature(gene, transcript, exonStore, leafFeatureStore);
+
+                if ( featureType.equals("gene") ) {
+                    decodedFeature = gene;
+                    break;
+                }
+                else if ( featureType.equals("transcript") ) {
+                    transcript = (GencodeGtfTranscriptFeature) feature;
+                    ++currentLineNum;
+                }
+            }
+            else {
+                // We have not reached the end of this set of gene / transcript records.
+                // We must cache these records together so we can create a meaningful data hierarchy from them all.
+                // Records are stored in their Feature form, not string form.
+
+                // Add the feature to the correct storage unit for easy assembly later:
+                switch (featureType) {
+                    case "gene":
+                        gene = (GencodeGtfGeneFeature)feature;
+                        break;
+                    case "transcript":
+                        transcript = (GencodeGtfTranscriptFeature)feature;
+                        break;
+                    case "exon":
+                        exonStore.add((GencodeGtfExonFeature)feature);
+                        break;
+                    default:
+                        leafFeatureStore.add(feature);
+                        break;
+                }
+
+                ++currentLineNum;
+            }
+
+            // Increment our iterator here so we don't accidentally miss any features from the following gene
+            lineIterator.next();
+        }
+
+        // Do we have some records leftover that we need to aggregate into a feature:
+        if ( (gene != null) &&
+                (exonStore.size() != 0) || (leafFeatureStore.size() != 0) ) {
+
+            aggregateRecordsIntoGeneFeature(gene, transcript, exonStore, leafFeatureStore);
+            decodedFeature = gene;
+        }
+
+        // If we have other records left over we should probably yell a lot,
+        // as this is bad.
+        //
+        // However, this should never actually happen.
+        //
+        if ( (exonStore.size() != 0) || (leafFeatureStore.size() != 0) ) {
+
+            if (exonStore.size() != 0) {
+                logger.error("Gene Feature Aggregation: Exon store not empty: " + exonStore.toString());
+            }
+
+            if (leafFeatureStore.size() != 0) {
+                logger.error("Gene Feature Aggregation: leaf feature store not empty: " + leafFeatureStore.toString());
+            }
+
+            String msg = "Aggregated data left over after parsing complete: Exons: " + exonStore.size() + " ; LeafFeatures: " + leafFeatureStore.size();
+            throw new RuntimeException(msg);
```

Comment:
@droazen Well... this case should never come up in practice since we check above and try to dump the remaining features into the gene.  I'm not sure there is a way for this to happen - should I just take the code out?